File: mycollab-services/src/main/java/com/esofthead/mycollab/module/project/dao/ProjectMemberMapperExt.java
/**
 * This file is part of mycollab-services.
 *
 * mycollab-services is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.esofthead.mycollab.module.project.dao;

import com.esofthead.mycollab.core.persistence.ISearchableDAO;
import com.esofthead.mycollab.module.project.domain.SimpleProjectMember;
import com.esofthead.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria;
import com.esofthead.mycollab.module.user.domain.SimpleUser;
import org.apache.ibatis.annotations.Param;

import java.util.List;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
public interface ProjectMemberMapperExt extends ISearchableDAO<ProjectMemberSearchCriteria> {
    SimpleProjectMember findMemberById(int memberId);

    List<SimpleUser> getUsersNotInProject(@Param("projectId") int projectId, @Param("sAccountId") int sAccountId);

    List<SimpleUser> getActiveUsersInProject(@Param("projectId") int projectId, @Param("sAccountId") int sAccountId);

    SimpleProjectMember findMemberByUsername(@Param("username") String username, @Param("projectId") int projectId);

    List<SimpleUser> getActiveUsersInProjects(@Param("projectIds") List<Integer> projectIds, @Param("sAccountId") Integer sAccountId);

    SimpleUser getActiveUserOfProject(@Param("username") String username, @Param("projectId") Integer projectId);
}

File: mycollab-services/src/main/java/com/mycollab/module/project/dao/ProjectMemberMapperExt.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.project.dao

import com.mycollab.db.persistence.ISearchableDAO
import com.mycollab.module.project.domain.SimpleProjectMember
import com.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria
import com.mycollab.module.user.domain.SimpleUser
import org.apache.ibatis.annotations.Param

import java.util.Date

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
interface ProjectMemberMapperExt : ISearchableDAO<ProjectMemberSearchCriteria> {
    fun findMemberById(memberId: Int): SimpleProjectMember

    fun getUsersNotInProject(@Param("projectId") projectId: Int, @Param("sAccountId") sAccountId: Int): List<SimpleUser>

    fun getActiveUsersInProject(@Param("projectId") projectId: Int, @Param("sAccountId") sAccountId: Int): List<SimpleUser>

    fun findMemberByUsername(@Param("username") username: String, @Param("projectId") projectId: Int): SimpleProjectMember

    fun getActiveUsersInProjects(@Param("projectIds") projectIds: List<Int>, @Param("sAccountId") sAccountId: Int?): List<SimpleUser>

    fun getActiveUserOfProject(@Param("username") username: String, @Param("projectId") projectId: Int?,
                               @Param("sAccountId") sAccountId: Int?): SimpleUser

    fun findMembersHourlyInProject(@Param("projectId") projectId: Int?, @Param("sAccountId") sAccountId: Int?,
                                   @Param("start") start: Date, @Param("end") end: Date): List<SimpleProjectMember>
}
